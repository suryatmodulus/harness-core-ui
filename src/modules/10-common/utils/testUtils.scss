/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.browser {
  border: 1px solid #cccccc;
  border-radius: 8px;
  min-height: 300px;
  overflow: auto;

  .browser-header {
    background-color: #cccccc;
    padding: 8px 16px;
    display: flex;
  }

  .browser-path {
    display: block;
    border: none;
    line-height: 14px;
    font-size: 12px;
    color: #444444;
    border-radius: 9px;
    min-width: 100px;
    padding: 2px 8px;
    flex: 1 1 auto;
    border: 1px solid transparent;

    &:focus {
      border-color: rgb(30, 167, 253);
      outline: none;
    }
  }

  .browser-content {
    padding: 8px;
  }
}
