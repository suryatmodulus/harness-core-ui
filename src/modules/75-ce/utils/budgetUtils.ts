/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

export enum AlertStatus {
  Danger,
  Warning,
  Info
}

interface CostStatusInfo {
  cost: number
  ratio: number
  percentage: number
  status: AlertStatus
}

export interface getAllBudgetCostInfoReturnType {
  actualCostStatus: CostStatusInfo
  forecastedCostStatus: CostStatusInfo
  budgetAmountStatus: CostStatusInfo
}

export const getAllBudgetCostInfo: (
  forecastedCost: number,
  actualCost: number,
  budgetAmount: number
) => getAllBudgetCostInfoReturnType = (forecastedCost, actualCost, budgetAmount) => {
  /**
   * Find out the max cost of all and it will be considered as 100%
   * and the rest of the cost percentages will be calculated based on that
   */

  const baseCost = Math.max(actualCost, forecastedCost, budgetAmount)
  const actualCostRatio = actualCost / baseCost
  const forecastCostRatio = forecastedCost / baseCost
  const budgetAmountRatio = budgetAmount / baseCost

  const actualCostPerWithBudget = (actualCost / budgetAmount) * 100
  const forecastCostPerWithBudget = (forecastedCost / budgetAmount) * 100

  const getAlertStatus: (percentage: number) => AlertStatus = percentage => {
    if (percentage > 100) {
      return AlertStatus.Danger
    }
    if (percentage > 90) {
      return AlertStatus.Warning
    }
    return AlertStatus.Info
  }

  const actualCostStatus: CostStatusInfo = {
    cost: actualCost,
    ratio: actualCostRatio,
    percentage: actualCostPerWithBudget,
    status: getAlertStatus(actualCostPerWithBudget)
  }
  const forecastedCostStatus: CostStatusInfo = {
    cost: forecastedCost,
    ratio: forecastCostRatio,
    percentage: forecastCostPerWithBudget,
    status: getAlertStatus(forecastCostPerWithBudget)
  }
  const budgetAmountStatus: CostStatusInfo = {
    cost: budgetAmount,
    ratio: budgetAmountRatio,
    percentage: 100,
    status: AlertStatus.Info
  }

  return {
    actualCostStatus,
    forecastedCostStatus,
    budgetAmountStatus
  }
}
