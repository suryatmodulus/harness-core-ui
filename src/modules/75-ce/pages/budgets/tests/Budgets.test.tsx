/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { Provider } from 'urql'
import { fromValue } from 'wonka'
import type { DocumentNode } from 'graphql'
import { render } from '@testing-library/react'
import { FetchBudgetDocument } from 'services/ce/services'
import { TestWrapper } from '@common/utils/testUtils'
import Budgets from '../Budgets'

import ResponseData from './BudgetList.json'

jest.mock('services/ce', () => ({
  useDeleteBudget: jest.fn().mockImplementation(() => ({
    mutate: async () => {
      return {
        status: 'SUCCESS',
        data: {}
      }
    }
  })),
  useGetPerspective: jest.fn().mockImplementation(() => ({
    mutate: async () => {
      return {
        status: 'SUCCESS',
        data: {}
      }
    }
  }))
}))

const params = {
  accountId: 'TEST_ACC',
  perspetiveId: 'perspectiveId',
  perspectiveName: 'sample perspective'
}

describe('test cases for Budgets List Page', () => {
  test('should be able to render the budget list page', async () => {
    const responseState = {
      executeQuery: ({ query }: { query: DocumentNode }) => {
        if (query === FetchBudgetDocument) {
          return fromValue(ResponseData)
        }
      }
    }

    const { container } = render(
      <TestWrapper pathParams={params}>
        <Provider value={responseState as any}>
          <Budgets />
        </Provider>
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
})
