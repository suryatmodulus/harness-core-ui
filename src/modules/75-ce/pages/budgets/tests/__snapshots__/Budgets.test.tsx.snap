// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test cases for Budgets List Page should be able to render the budget list page 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main PageHeader--container PageHeader--standard StyledProps--flex StyledProps--padding-left-xlarge StyledProps--padding-right-xlarge StyledProps--background-white"
  >
    <div
      class="Layout--vertical StyledProps--main"
    >
      <h2
        class="StyledProps--font StyledProps--font-h2 StyledProps--main StyledProps--color StyledProps--color-grey800 StyledProps--font-weight-bold"
        level="2"
      >
        ce.budgets.listPage.title
      </h2>
    </div>
  </div>
  <div
    class="StyledProps--font StyledProps--main PageBody--pageBody"
    data-name="page-body"
  >
    <div
      class="Layout--horizontal StyledProps--main StyledProps--padding-left-large StyledProps--padding-right-large StyledProps--padding-top-medium StyledProps--padding-bottom-medium StyledProps--background-white"
    >
      <button
        class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color Button--withLeftIcon"
        type="button"
      >
        <span
          class="bp3-icon bp3-icon-plus StyledProps--main StyledProps--padding-right-xsmall"
          icon="plus"
        >
          <svg
            data-icon="plus"
            height="10"
            viewBox="0 0 16 16"
            width="10"
          >
            <desc>
              plus
            </desc>
            <path
              d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        <span
          class="bp3-button-text"
        >
          ce.budgets.listPage.newBudget
        </span>
      </button>
      <span
        style="flex-grow: 1;"
      />
      <div
        class="bp3-input-group ui-search-box ExpandingSearchInput--main search   ExpandingSearchInput--light"
        data-name=""
      >
        <span
          class="bp3-icon StyledProps--main ExpandingSearchInput--icon StyledProps--color StyledProps--color-grey700"
          data-icon="thinner-search"
        >
          <svg
            fill="none"
            height="14"
            viewBox="0 0 16 16"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.125 13.25c1.617 0 3.32-.695 4.375-1.75L15 15l-3.5-3.5c1.021-1.005 1.75-2.894 1.75-4.375a6.125 6.125 0 10-6.125 6.125z"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </span>
        <input
          class="bp3-input  ExpandingSearchInput--filter"
          dir="auto"
          placeholder="ce.budgets.listPage.searchText"
          style="padding-right: 0px;"
          type="search"
          value=""
        />
      </div>
    </div>
    <div
      class="Layout--horizontal StyledProps--main StyledProps--padding-large"
    >
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-variation-h5"
      >
        ce.budgets.listPage.budgetCount
      </p>
    </div>
    <div
      class="StyledProps--font StyledProps--main StyledProps--padding-large"
    >
      <div
        class="TableV2--table"
      >
        <div
          class="TableV2--header"
          role="row"
        >
          <div
            class="TableV2--cell"
            colspan="1"
            role="columnheader"
            style="width: 25%;"
          >
            <p
              class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
            >
              ce.budgets.listPage.tableHeaders.name
            </p>
          </div>
          <div
            class="TableV2--cell"
            colspan="1"
            role="columnheader"
            style="width: 20%;"
          >
            <p
              class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
            >
              ce.budgets.listPage.tableHeaders.budgetAmount
            </p>
          </div>
          <div
            class="TableV2--cell"
            colspan="1"
            role="columnheader"
            style="width: 35%;"
          >
            <p
              class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
            >
              ce.budgets.listPage.tableHeaders.monthToDate
            </p>
          </div>
          <div
            class="TableV2--cell"
            colspan="1"
            role="columnheader"
            style="width: 15%;"
          >
            <p
              class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
            >
              ce.budgets.listPage.tableHeaders.alerts
            </p>
          </div>
          <div
            class="TableV2--cell"
            colspan="1"
            role="columnheader"
            style="width: 5%;"
          >
            <p
              class="StyledProps--font StyledProps--main StyledProps--font-variation-table-headers"
            >
               
            </p>
          </div>
        </div>
        <div
          class="TableV2--body"
        >
          <div
            class="TableV2--row TableV2--card TableV2--clickable"
            role="row"
          >
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 25%;"
            >
              <div
                class="Layout--vertical StyledProps--main"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-black"
                >
                  App budget
                </p>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-variation-small StyledProps--color StyledProps--color-grey600"
                >
                  ce.budgets.listPage.timePeriodRemaining
                </p>
              </div>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 20%;"
            >
              <p
                class="StyledProps--font StyledProps--main StyledProps--font-variation-h5 StyledProps--color StyledProps--color-grey800"
              >
                $10.00
              </p>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 35%;"
            >
              <div
                class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main StyledProps--padding-top-medium"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400 StyledProps--font-variation-tiny"
                >
                  $0.00
                </p>
                <div
                  class="StyledProps--font StyledProps--main budgetBar StyledProps--background-grey200"
                  style="width: 266px;"
                >
                  <div
                    class="actualCostBar"
                    style="width: 73px;"
                  />
                  <div
                    class="forecastedBar"
                    style="width: 212px;"
                  />
                  <div
                    class="actualCostMarker"
                    style="left: 73px;"
                  />
                  <p
                    class="StyledProps--font StyledProps--main actualCost StyledProps--font-variation-small-bold StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
                    style="left: 73px;"
                  >
                    $2.77
                  </p>
                  <div
                    class="overBudgetMarker"
                    style="left: 212px; opacity: 0;"
                  />
                  <p
                    class="StyledProps--font StyledProps--main overBudgetText StyledProps--font-variation-small-bold StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall StyledProps--color StyledProps--color-red700"
                    style="left: 212px; opacity: 0; display: flex; align-items: center;"
                  >
                    <span
                      class="bp3-icon StyledProps--main Text--icon StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-red500"
                      data-icon="execution-warning"
                    >
                      <svg
                        height="14"
                        viewBox="0 0 19 15"
                        width="14"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          color="#fff"
                          d="M15.964 15H3.027C.702 15-.737 12.787.392 11.021L6.29 1.339c1.152-1.785 5.261-1.785 6.412 0l5.897 9.682C19.75 12.768 18.288 15 15.964 15z"
                        />
                        <path
                          d="M4.23 13.75h10.408c1.713 0 2.79-1.77 1.941-3.157l-5.22-8.53c-.848-1.417-3.002-1.417-3.85 0l-5.22 8.53c-.832 1.402.229 3.157 1.941 3.157zm5.22-1.894c-.456 0-.832-.354-.832-.785 0-.431.376-.786.832-.786.457 0 .832.354.816.801.016.416-.375.77-.816.77zm-.212-7.083c.392-.093.8.092.963.446.049.123.081.247.081.385-.016.385-.049.785-.081 1.17-.033.601-.082 1.217-.114 1.817-.017.2-.017.37-.017.57a.6.6 0 01-.62.57c-.342 0-.603-.231-.62-.554-.048-.94-.114-1.879-.163-2.818l-.049-.755a.82.82 0 01.62-.831z"
                        />
                      </svg>
                    </span>
                    ce.budgets.listPage.overBudgetText
                  </p>
                </div>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400 StyledProps--font-variation-tiny"
                >
                  $10.00
                </p>
              </div>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 15%;"
            >
              <div
                class="Layout--vertical StyledProps--main"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-variation-small StyledProps--color StyledProps--color-grey600"
                >
                  ce.budgets.listPage.mtdAlert
                </p>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-variation-small StyledProps--color StyledProps--color-grey600"
                >
                  ce.budgets.listPage.forecastAlert
                </p>
              </div>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 5%;"
            >
              <span
                class="bp3-popover-wrapper bp3-dark"
              >
                <span
                  class="bp3-popover-target"
                >
                  <button
                    class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
                    type="button"
                  >
                    <span
                      class="bp3-icon StyledProps--main"
                      data-icon="Options"
                    >
                      <svg
                        fill="none"
                        height="16"
                        viewBox="0 0 4 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M2 12.5a1 1 0 100 2 1 1 0 000-2zm-2 1a2 2 0 114 0 2 2 0 01-4 0zm2-12a1 1 0 100 2 1 1 0 000-2zm-2 1a2 2 0 114 0 2 2 0 01-4 0zM2 7a1 1 0 100 2 1 1 0 000-2zM0 8a2 2 0 114 0 2 2 0 01-4 0z"
                          fill="currentColor"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </button>
                </span>
              </span>
            </div>
          </div>
          <div
            class="TableV2--row TableV2--card TableV2--clickable"
            role="row"
          >
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 25%;"
            >
              <div
                class="Layout--vertical StyledProps--main"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-black"
                >
                  AzureBudget
                </p>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-variation-small StyledProps--color StyledProps--color-grey600"
                >
                  ce.budgets.listPage.timePeriodRemaining
                </p>
              </div>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 20%;"
            >
              <p
                class="StyledProps--font StyledProps--main StyledProps--font-variation-h5 StyledProps--color StyledProps--color-grey800"
              >
                $80,000.00
              </p>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 35%;"
            >
              <div
                class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main StyledProps--padding-top-medium"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400 StyledProps--font-variation-tiny"
                >
                  $0.00
                </p>
                <div
                  class="StyledProps--font StyledProps--main budgetBar StyledProps--background-grey200"
                  style="width: 266px;"
                >
                  <div
                    class="actualCostBar"
                    style="width: 184px;"
                  />
                  <div
                    class="forecastedBar"
                    style="width: 199px;"
                  />
                  <div
                    class="actualCostMarker"
                    style="left: 184px;"
                  />
                  <p
                    class="StyledProps--font StyledProps--main actualCost left StyledProps--font-variation-small-bold StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
                    style="left: 184px;"
                  >
                    $55,428.46
                  </p>
                  <div
                    class="overBudgetMarker"
                    style="left: 199px; opacity: 0;"
                  />
                  <p
                    class="StyledProps--font StyledProps--main overBudgetText StyledProps--font-variation-small-bold StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall StyledProps--color StyledProps--color-red700"
                    style="left: 199px; opacity: 0; display: flex; align-items: center;"
                  >
                    <span
                      class="bp3-icon StyledProps--main Text--icon StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-red500"
                      data-icon="execution-warning"
                    >
                      <svg
                        height="14"
                        viewBox="0 0 19 15"
                        width="14"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          color="#fff"
                          d="M15.964 15H3.027C.702 15-.737 12.787.392 11.021L6.29 1.339c1.152-1.785 5.261-1.785 6.412 0l5.897 9.682C19.75 12.768 18.288 15 15.964 15z"
                        />
                        <path
                          d="M4.23 13.75h10.408c1.713 0 2.79-1.77 1.941-3.157l-5.22-8.53c-.848-1.417-3.002-1.417-3.85 0l-5.22 8.53c-.832 1.402.229 3.157 1.941 3.157zm5.22-1.894c-.456 0-.832-.354-.832-.785 0-.431.376-.786.832-.786.457 0 .832.354.816.801.016.416-.375.77-.816.77zm-.212-7.083c.392-.093.8.092.963.446.049.123.081.247.081.385-.016.385-.049.785-.081 1.17-.033.601-.082 1.217-.114 1.817-.017.2-.017.37-.017.57a.6.6 0 01-.62.57c-.342 0-.603-.231-.62-.554-.048-.94-.114-1.879-.163-2.818l-.049-.755a.82.82 0 01.62-.831z"
                        />
                      </svg>
                    </span>
                    ce.budgets.listPage.overBudgetText
                  </p>
                </div>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400 StyledProps--font-variation-tiny"
                >
                  $80,000.00
                </p>
              </div>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 15%;"
            >
              <div
                class="Layout--vertical StyledProps--main"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-variation-small StyledProps--color StyledProps--color-grey600"
                >
                  ce.budgets.listPage.mtdAlert
                </p>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-variation-small StyledProps--color StyledProps--color-grey600"
                >
                  ce.budgets.listPage.forecastAlert
                </p>
              </div>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 5%;"
            >
              <span
                class="bp3-popover-wrapper bp3-dark"
              >
                <span
                  class="bp3-popover-target"
                >
                  <button
                    class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
                    type="button"
                  >
                    <span
                      class="bp3-icon StyledProps--main"
                      data-icon="Options"
                    >
                      <svg
                        fill="none"
                        height="16"
                        viewBox="0 0 4 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M2 12.5a1 1 0 100 2 1 1 0 000-2zm-2 1a2 2 0 114 0 2 2 0 01-4 0zm2-12a1 1 0 100 2 1 1 0 000-2zm-2 1a2 2 0 114 0 2 2 0 01-4 0zM2 7a1 1 0 100 2 1 1 0 000-2zM0 8a2 2 0 114 0 2 2 0 01-4 0z"
                          fill="currentColor"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </button>
                </span>
              </span>
            </div>
          </div>
          <div
            class="TableV2--row TableV2--card TableV2--clickable"
            role="row"
          >
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 25%;"
            >
              <div
                class="Layout--vertical StyledProps--main"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-black"
                >
                  sv azure 1
                </p>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-variation-small StyledProps--color StyledProps--color-grey600"
                >
                  ce.budgets.listPage.timePeriodRemaining
                </p>
              </div>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 20%;"
            >
              <p
                class="StyledProps--font StyledProps--main StyledProps--font-variation-h5 StyledProps--color StyledProps--color-grey800"
              >
                $10,000.00
              </p>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 35%;"
            >
              <div
                class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main StyledProps--padding-top-medium"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400 StyledProps--font-variation-tiny"
                >
                  $0.00
                </p>
                <div
                  class="StyledProps--font StyledProps--main budgetBar StyledProps--background-grey200"
                  style="width: 266px;"
                >
                  <div
                    class="actualCostBar"
                    style="width: 3px;"
                  />
                  <div
                    class="forecastedBar danger"
                    style="width: 266px;"
                  />
                  <div
                    class="budgetStatusBar"
                    style="width: 44px;"
                  />
                  <div
                    class="actualCostMarker"
                    style="left: 3px;"
                  />
                  <p
                    class="StyledProps--font StyledProps--main actualCost StyledProps--font-variation-small-bold StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
                    style="left: 3px;"
                  >
                    $866.68
                  </p>
                  <div
                    class="overBudgetMarker"
                    style="left: 266px; opacity: 1;"
                  />
                  <p
                    class="StyledProps--font StyledProps--main overBudgetText StyledProps--font-variation-small-bold StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall StyledProps--color StyledProps--color-red700"
                    style="left: 266px; opacity: 1; display: flex; align-items: center;"
                  >
                    <span
                      class="bp3-icon StyledProps--main Text--icon StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-red500"
                      data-icon="execution-warning"
                    >
                      <svg
                        height="14"
                        viewBox="0 0 19 15"
                        width="14"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          color="#fff"
                          d="M15.964 15H3.027C.702 15-.737 12.787.392 11.021L6.29 1.339c1.152-1.785 5.261-1.785 6.412 0l5.897 9.682C19.75 12.768 18.288 15 15.964 15z"
                        />
                        <path
                          d="M4.23 13.75h10.408c1.713 0 2.79-1.77 1.941-3.157l-5.22-8.53c-.848-1.417-3.002-1.417-3.85 0l-5.22 8.53c-.832 1.402.229 3.157 1.941 3.157zm5.22-1.894c-.456 0-.832-.354-.832-.785 0-.431.376-.786.832-.786.457 0 .832.354.816.801.016.416-.375.77-.816.77zm-.212-7.083c.392-.093.8.092.963.446.049.123.081.247.081.385-.016.385-.049.785-.081 1.17-.033.601-.082 1.217-.114 1.817-.017.2-.017.37-.017.57a.6.6 0 01-.62.57c-.342 0-.603-.231-.62-.554-.048-.94-.114-1.879-.163-2.818l-.049-.755a.82.82 0 01.62-.831z"
                        />
                      </svg>
                    </span>
                    ce.budgets.listPage.overBudgetText
                  </p>
                </div>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-red800 StyledProps--font-variation-tiny"
                >
                  $60,000.00
                </p>
              </div>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 15%;"
            >
              <div
                class="Layout--vertical StyledProps--main"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-variation-small StyledProps--color StyledProps--color-grey600"
                >
                  ce.budgets.listPage.mtdAlert
                </p>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-variation-small StyledProps--color StyledProps--color-grey600"
                >
                  ce.budgets.listPage.forecastAlert
                </p>
              </div>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 5%;"
            >
              <span
                class="bp3-popover-wrapper bp3-dark"
              >
                <span
                  class="bp3-popover-target"
                >
                  <button
                    class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
                    type="button"
                  >
                    <span
                      class="bp3-icon StyledProps--main"
                      data-icon="Options"
                    >
                      <svg
                        fill="none"
                        height="16"
                        viewBox="0 0 4 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M2 12.5a1 1 0 100 2 1 1 0 000-2zm-2 1a2 2 0 114 0 2 2 0 01-4 0zm2-12a1 1 0 100 2 1 1 0 000-2zm-2 1a2 2 0 114 0 2 2 0 01-4 0zM2 7a1 1 0 100 2 1 1 0 000-2zM0 8a2 2 0 114 0 2 2 0 01-4 0z"
                          fill="currentColor"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </button>
                </span>
              </span>
            </div>
          </div>
          <div
            class="TableV2--row TableV2--card TableV2--clickable"
            role="row"
          >
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 25%;"
            >
              <div
                class="Layout--vertical StyledProps--main"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-black"
                >
                  sample
                </p>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-variation-small StyledProps--color StyledProps--color-grey600"
                >
                  ce.budgets.listPage.timePeriodRemaining
                </p>
              </div>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 20%;"
            >
              <p
                class="StyledProps--font StyledProps--main StyledProps--font-variation-h5 StyledProps--color StyledProps--color-grey800"
              >
                $10.00
              </p>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 35%;"
            >
              <div
                class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main StyledProps--padding-top-medium"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400 StyledProps--font-variation-tiny"
                >
                  $0.00
                </p>
                <div
                  class="StyledProps--font StyledProps--main budgetBar StyledProps--background-grey200"
                  style="width: 266px;"
                >
                  <div
                    class="actualCostBar danger"
                    style="width: 144px;"
                  />
                  <div
                    class="forecastedBar danger"
                    style="width: 266px;"
                  />
                  <div
                    class="budgetStatusBar"
                    style="width: 53px;"
                  />
                  <div
                    class="actualCostMarker danger"
                    style="left: 144px;"
                  />
                  <p
                    class="StyledProps--font StyledProps--main actualCost left StyledProps--font-variation-small-bold StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall"
                    style="left: 144px;"
                  >
                    $27.14
                  </p>
                  <div
                    class="overBudgetMarker"
                    style="left: 266px; opacity: 1;"
                  />
                  <p
                    class="StyledProps--font StyledProps--main overBudgetText StyledProps--font-variation-small-bold StyledProps--padding-right-xsmall StyledProps--padding-left-xsmall StyledProps--color StyledProps--color-red700"
                    style="left: 266px; opacity: 1; display: flex; align-items: center;"
                  >
                    <span
                      class="bp3-icon StyledProps--main Text--icon StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-red500"
                      data-icon="execution-warning"
                    >
                      <svg
                        height="14"
                        viewBox="0 0 19 15"
                        width="14"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          color="#fff"
                          d="M15.964 15H3.027C.702 15-.737 12.787.392 11.021L6.29 1.339c1.152-1.785 5.261-1.785 6.412 0l5.897 9.682C19.75 12.768 18.288 15 15.964 15z"
                        />
                        <path
                          d="M4.23 13.75h10.408c1.713 0 2.79-1.77 1.941-3.157l-5.22-8.53c-.848-1.417-3.002-1.417-3.85 0l-5.22 8.53c-.832 1.402.229 3.157 1.941 3.157zm5.22-1.894c-.456 0-.832-.354-.832-.785 0-.431.376-.786.832-.786.457 0 .832.354.816.801.016.416-.375.77-.816.77zm-.212-7.083c.392-.093.8.092.963.446.049.123.081.247.081.385-.016.385-.049.785-.081 1.17-.033.601-.082 1.217-.114 1.817-.017.2-.017.37-.017.57a.6.6 0 01-.62.57c-.342 0-.603-.231-.62-.554-.048-.94-.114-1.879-.163-2.818l-.049-.755a.82.82 0 01.62-.831z"
                        />
                      </svg>
                    </span>
                    ce.budgets.listPage.overBudgetText
                  </p>
                </div>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-red800 StyledProps--font-variation-tiny"
                >
                  $50.00
                </p>
              </div>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 15%;"
            >
              <div
                class="Layout--vertical StyledProps--main"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-variation-small StyledProps--color StyledProps--color-grey600"
                >
                  ce.budgets.listPage.mtdAlert
                </p>
                <p
                  class="StyledProps--font StyledProps--main StyledProps--font-variation-small StyledProps--color StyledProps--color-grey600"
                >
                  ce.budgets.listPage.forecastAlert
                </p>
              </div>
            </div>
            <div
              class="TableV2--cell"
              role="cell"
              style="width: 5%;"
            >
              <span
                class="bp3-popover-wrapper bp3-dark"
              >
                <span
                  class="bp3-popover-target"
                >
                  <button
                    class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--without-shadow Button--withLeftIcon"
                    type="button"
                  >
                    <span
                      class="bp3-icon StyledProps--main"
                      data-icon="Options"
                    >
                      <svg
                        fill="none"
                        height="16"
                        viewBox="0 0 4 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M2 12.5a1 1 0 100 2 1 1 0 000-2zm-2 1a2 2 0 114 0 2 2 0 01-4 0zm2-12a1 1 0 100 2 1 1 0 000-2zm-2 1a2 2 0 114 0 2 2 0 01-4 0zM2 7a1 1 0 100 2 1 1 0 000-2zM0 8a2 2 0 114 0 2 2 0 01-4 0z"
                          fill="currentColor"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </span>
                  </button>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
